### PR TITLE
Docker spark connectors

### LINKED ISSUE #2471: Integration tests for Spark-family of connectors.

### PR DESCRIPTION
**Pull request by nicflores**
Add spark connector infrastructure for integration testing. Fixes #2471, #2414, #2413, #2412, #2411, #2410.

### CODE REVIEW

File: docker/scripts/setupContainers [hunk 1]
```diff
@@ -42,23 +42,45 @@ init_marklogic() {
...
     mkdir ${HOME}/spark_local_test
-    ls ${HOME}
 }
 
+init_hdfs() {
+  # Need to start datanode and nodemanager
+  # within hadoop container
+  CONTAINER=$1
+  echo "configuring hdfs..."
+  docker exec $CONTAINER /hadoop-2.7.2/sbin/hadoop-daemon.sh start datanode
```
Comment (rabbitonweb) [inline docker/scripts/setupContainers]: we need 2.7.3 here

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -1,65 +1,130 @@
...
+      - "21:21"
+      - "30000-30009:30000-30009"
+    environment:
+      - PUBLICHOST=0.0.0.0
+    volumes:
+      - ./Dockerfiles/FTP:/etc/pure-ftpd/passwd
+  
+  quasar_spark_hdfs:
+    container_name: quasar_spark_hdfs
+    image: harisekhon/hadoop
```
Comment (rabbitonweb) [inline docker/docker-compose.yml]: do we know which version of hdfs this image is running?

Comment (nicflores) [inline docker/docker-compose.yml]: @rabbitonweb , it uses 2.7.2

File: docker/scripts/setupContainers (hunk 1 above)
Comment (nicflores) [inline docker/scripts/setupContainers]: I'll look into upgrading or finding another image which uses 2.7.3.

**Pull request merged**